src/Tests/TestBuilder.ts

```diff
@@ -0,0 +1,135 @@
+  }
+
+  setUp() {
+    try {
+      if (fs.existsSync(this.dirInTemp)) {
+        fs.rmdirSync(this.dirInTemp, {recursive: true});
+      }
+
+      if (fs.existsSync(this.dirInWorkspace)) {
+        fs.rmdirSync(this.dirInWorkspace, {recursive: true});
+      }
+
+      fs.mkdirSync(this.dirInTemp, {recursive: true});
+      fs.mkdirSync(this.dirInWorkspace, {recursive: true});
+      console.log(`Test ${this.testLabel} - Start`);
+    } catch (e) {
+      console.error('Cannot create temporal directory for the test');
+      throw e;
+    }
+  }
+
+  getPath(fileName: string, tempOrWorkspace: string = 'temp') {
+    if (tempOrWorkspace === 'temp') {
+      return `${this.dirInTemp}/${fileName}`;
+    } else if (tempOrWorkspace === 'workspace') {
+      return `${this.dirInWorkspace}/${fileName}`;
+    } else {
+      throw Error('Invalid parameter');
+    }
+  }
+
+  writeFileSync(fileName: string, content: string, tempOrWorkspace: string = 'temp') {
+    const filePath = this.getPath(fileName, tempOrWorkspace);
+
+    try {
+      if (!fs.existsSync(path.dirname(filePath))) {
+        fs.mkdirSync(path.dirname(filePath));
+      }
+      fs.writeFileSync(filePath, content, 'utf-8');
+      console.log(`Test file is created (${filePath})`);
+    } catch (e) {
+      console.error('Cannot create temporal files for the test');
+      throw e;
+    }
+  }
+
+  tearDown() {
+    try {
+      if (fs.existsSync(this.dirInTemp)) {
+        fs.rmdirSync(this.dirInTemp, {recursive: true});
+      }
+      if (fs.existsSync(this.dirInTemp)) {
+        fs.rmdirSync(this.dirInWorkspace, {recursive: true});
+      }
+
+    } catch (e) {
+      // Do not throw to proceed the test
+      console.error('Cannot remove the test directory');
+    } finally {
+      console.log(`Test ${this.testLabel} - Done`);
+    }
+  }
+
+  static setUp() {
+    try {
+      if (fs.existsSync(TestBuilder.tempDir)) {
+        fs.rmdirSync(TestBuilder.tempDir, {recursive: true});
+      }
+      if (fs.existsSync(TestBuilder.workspaceDir)) {
+        fs.rmdirSync(TestBuilder.workspaceDir, {recursive: true});
+      }
+
+      fs.mkdirSync(TestBuilder.tempDir, {recursive: true});
+      fs.mkdirSync(TestBuilder.workspaceDir, {recursive: true});
+    } catch (e) {
+      console.error('Cannot create temporal directory for the test');
+      throw e;
+    }
+  }
+
+  static tearDown() {
+    try {
+      fs.rmdirSync(TestBuilder.tempDir, {recursive: true});
+      console.log(`Test directory is removed successfully. (${TestBuilder.tempDir})`);
+      fs.rmdirSync(TestBuilder.workspaceDir, {recursive: true});
+      console.log(
+          `Test directory in worksp is removed successfully. (${TestBuilder.workspaceDir})`);
+    } catch (e) {
+      // Do not throw to proceed the test
+      console.error('Cannot remove the test directory');
+    } finally {
+      console.log(`Removed ${TestBuilder.tempDir} and ${TestBuilder.tempDir}.`);
```

Comment:
These seems not matched each other :(
If an error is caught in `catch`, it means that the directories are not removed.

Reply:
Oh, sure. Thanks.